lib/cozy-light.js

```diff
@@ -1,1296 +1,93 @@
-      } else {
-        LOGGER.info(plugin + ' installed. Enjoy!');
-        if (configHelpers.mainWatcher !== undefined) {
-          configWatcher.one(callback);
-        }
-        configHelpers.addPlugin(plugin, manifest);
-        pluginHelpers.loadPlugin(plugin);
-        if (configHelpers.mainWatcher === undefined
-            && callback !== undefined && typeof(callback) === 'function') {
-          callback();
-        }
-      }
-    });
-  },
-
-  /**
- * Remove plugin from config and its source from node module folder.
-   *
-   * @param {String} plugin Plugin to remove.
-   * @param {Function} callback Termination.
-   */
-  uninstallPlugin: function (plugin, callback){
-    LOGGER.info('Removing ' + plugin + '...');
-    if (config.plugins[plugin] === undefined) {
-      LOGGER.error(plugin + ' is not installed.');
-    } else {
-      var module = config.plugins[plugin].name;
-      npmHelpers.uninstall(module, function remotePluginFromConfig (err) {
-        if (err) {
-          LOGGER.raw(err);
-          LOGGER.error('npm did not uninstall ' + plugin + ' correctly.');
-          callback(err);
-        } else {
-          LOGGER.info(plugin + ' successfully uninstalled.');
-          delete loadedPlugins[plugin];
-
-          if (configHelpers.mainWatcher !== undefined) {
-            configWatcher.one(callback);
-          }
-          configHelpers.removePlugin(plugin);
-          if (configHelpers.mainWatcher === undefined
-              && callback !== undefined && typeof(callback) === 'function') {
-            callback();
-          }
-        }
-      });
-    }
-  },
-
-  installDistro: function (distro, callback) {
-    if (distro !== undefined && distro !== '') {
-      try {
-        LOGGER.info('Start distribution installation for ' + distro);
-        distrib.installDistro(distro, actions, function(err) {
-          if (err) {
-            LOGGER.raw(err.message);
-            LOGGER.error(
-              'An error occured while installing your distribution');
-            if (callback !== undefined && typeof(callback) === 'function') {
-              callback(err);
-            }
-          } else {
-            LOGGER.info('Distribution ' + distro + 'successfully installed');
-            if (callback !== undefined && typeof(callback) === 'function') {
-              callback();
-            }
-          }
-        });
-      } catch (err) {
-        console.log(err);
-        LOGGER.error('An error occured while installing your distribution');
-      }
-    } else {
-      LOGGER.info(
-        'install-distro requires a distro name, here are the ones avalaible:');
-      distrib.displayDistros();
-    }
-  },
-
-  /**
-   * Display configuration file contents: apps configuration and user settings.
-   */
-  displayConfig: function () {
-    LOGGER.raw(JSON.stringify(configHelpers.loadConfigFile(), null, 2));
-  }
+  logger: printit({ prefix: 'Cozy Light' }),
+  db: null,
+  routes: {},
+  configHelpers: {},
```

Comment:
```
+ nodeHelpers: {},
+ npmHelpers: {},
+ pluginHelpers: {},
+ applicationHelpers: {},
+ mainAppHelper: {},
+ actions: {}
};
+cozyLight.configHelpers = require('./config-helper')();
+cozyLight.nodeHelpers = require('./node-helpers');
+cozyLight.npmHelpers = require('./npm-helpers')(cozyLight);
+cozyLight.pluginHelpers = require('./plugin-helpers')(cozyLight);
+cozyLight.applicationHelpers = require('./application-helpers')(cozyLight);
+cozyLight.mainAppHelper = require('./main-app-helper')(cozyLight);
+cozyLight.actions = require('./actions')(cozyLight);
```

All that attributes. I think they could be assigned directly instead of doing it after.